src/sessions.rs

```diff
@@ -3,18 +3,44 @@ use glib::warn;
 use glob::glob;
 use gtk::gio::DesktopAppInfo;
 use gtk::prelude::*;
-use std::collections::HashMap;
+use lazy_static::lazy_static;
+use std::{
+  collections::HashMap,
+  path::{PathBuf},
+  env,
+};
 
 static G_LOG_DOMAIN: &str = "phrog-sessions";
 
+lazy_static! {
+    // Snippet copied from https://github.com/apognu/tuigreet
+
+    static ref XDG_DATA_DIRS: Vec<PathBuf> = {
+        let value = env::var("XDG_DATA_DIRS").unwrap_or("/usr/local/share:/usr/share".to_string());
+        env::split_paths(&value).filter(|p| p.is_absolute()).collect()
+    };
+}
+
 pub fn sessions() -> Vec<SessionObject> {
     let mut sessions = HashMap::new();
-    session_list(
-        "/usr/share/wayland-sessions/*.desktop",
-        "wayland",
-        &mut sessions,
-    );
-    session_list("/usr/share/xsessions/*.desktop", "x11", &mut sessions);
+
+    for dir in XDG_DATA_DIRS.iter() {
+        let wl_dir = dir.join("wayland-sessions/*.desktop");
+        let x11_dir = dir.join("wayland-sessions/*.desktop");
+
+        session_list(
+            &wl_dir.into_os_string().into_string().unwrap(),
```

Comment:
```suggestion
            &wl_dir.to_str().unwrap(),
```

Same for the x11 part below, however it might be interesting to modify `session_list()` so it takes a `PathBuf` rather than a `&str`.

Reply:
I actually thought about it but i wanted to make minimal code changes since Rust is so vastly different from anything that resembles C programming

Reply:
Ok, so this uses glob so i dunno

Reply:
If making it a `PathBuf` isn't an option, then let's keep things as is (although my suggestion above still applies), moving from `&str` to `OsStr` wouldn't bring any benefit.

Reply:
I'd like to see `session_list(path: &PathBuf...` as @a-wai suggested :+1: